app/components/containers/containersController.js

```diff
@@ -239,17 +241,23 @@ angular.module('containers', [])
         return;
       }
     });
+    var title = 'You are about to remove one or more container.';
     if (isOneContainerRunning) {
-      ModalService.confirmDeletion(
-        'You are about to remove one or more running containers.',
-        function (confirmed) {
-          if(!confirmed) { return; }
-          $scope.removeAction();
-        }
-      );
-    } else {
-      $scope.removeAction();
+      title = 'You are about to remove one or more running containers.';
     }
+    var text = '<label for="clean" class="control-label text-left">Automatically clean associated volumes</label><label class="switch" style="margin-left: 21px;"><input type="checkbox"><i></i></label>';
+    ModalService.confirmContainerDeletion(
+      title,
+      text,
+      function (result) {
+        if(!result) { return; }
+        $scope.cleanAssociatedVolumes = false;
```

Comment:
Just tested, I add that because of the following scenario : 
- You list containers
- You select 2 containers and remove
- You choose to remove associated volumes
- Container list refresh (but not page - so value of scope stay at true)
- You select 2 containers and remove
- You choose to *not* remove associated volumes
=> Associated volumes are removed...